### PR TITLE
Business model

### PR DESCRIPTION
**Pull request by nirmal864**
#36 added business model
If there are any missing types for business, client or suppliers, please let me know, thanks

### CODE REVIEW

File: Client/package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "@testing-library/react": "^11.2.5",
     "@testing-library/user-event": "^12.6.3",
     "axios": "^0.21.1",
+    "express": "^4.17.1",
```
Comment (Pyxsys) [inline Client/package.json]: Express should not exist in client directory

File: Client/src/Forms/validateLogin.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ export default function validateLogin(values){
         errors.password = "Password required"
     }
     else{
-        console.log("No Errors")
+        console.log("No Err ors")
```
Comment (Pyxsys) [inline Client/src/Forms/validateLogin.js]: fix typo

File: Server/src/Models/Client.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+
+const mongoose = require('mongoose');
+const business_schema = require('./Business.js');
+
+  // Schema
+  const client_schema = mongoose.Schema({
+    _id: { type: Number},
```
Comment (Pyxsys) [inline Server/src/Models/Client.js]: Ref id's are automatically generated by mongo. Did you mean to use an internal ID?

File: Server/src/Models/Supplier.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+const business_schema = require('./Business.js');
+
+
+// Schema
+const supplier_schema = mongoose.Schema({
+    _id: { type: Number},
+    name:       {type: String, required: true},
+    phone:       {type: String, required: true},
+    location:          {type: String, required: true},
+    businessArr: [business_schema]
```
Comment (Pyxsys) [inline Server/src/Models/Supplier.js]: Causes 'name' field to be redundant. What is the intention behind this?

File: Server/src/Models/Supplier.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose');
+const business_schema = require('./Business.js');
+
+
+// Schema
+const supplier_schema = mongoose.Schema({
+    _id: { type: Number},
```
Comment (Pyxsys) [inline Server/src/Models/Supplier.js]: Ref id's are generated automatically by mongo

File: Server/src/Models/Client.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    name:       {type: String, required: true},
+    phone:       {type: String, required: true},
+    address:          {type: String, required: true},
+    businessArr: [business_schema]
```
Comment (Pyxsys) [inline Server/src/Models/Client.js]: Causes 'name' field to be redundant. What is the intention behind this?

File: Server/src/Models/Business.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+const mongoose = require('mongoose');
+
+// Schema
+const business_schema = mongoose.Schema({
+    name: {type: String, required: true}
```
Comment (Pyxsys) [inline Server/src/Models/Business.js]: Parent schema should hold more global/general info that applies to all business entities

File: Server/src/Models/Client.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+
+const mongoose = require('mongoose');
+const business_schema = require('./Business.js');
+
+  // Schema
+  const client_schema = mongoose.Schema({
+    _id: { type: Number},
+    name:       {type: String, required: true},
+    phone:       {type: String, required: true},
+    address:          {type: String, required: true},
```
Comment (Pyxsys) [inline Server/src/Models/Client.js]: Duplicated code on Supplier.js

File: Server/src/Models/Supplier.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    name:       {type: String, required: true},
+    phone:       {type: String, required: true},
+    location:          {type: String, required: true},
```
Comment (Pyxsys) [inline Server/src/Models/Supplier.js]: Duplicated code on Client.js

**Review by Pyxsys — CHANGES_REQUESTED** [review]
- `Client/` changes should be reverted/fixed.

- Models appear to not use inheritance principles while employing them, leading to duplicate code.

Please refer to the comments made.
Please merge back-end task branches into BE_PreProd and not into main form here on out.
It'll only require the review of one person as opposed to three.

File: Client/package.json (hunk 1 above)
Comment (nirmal864) [reply Client/package.json]: removed

File: Client/src/Forms/validateLogin.js (hunk 2 above)
Comment (nirmal864) [reply Client/src/Forms/validateLogin.js]: fixed

File: Server/src/Models/Client.js (hunk 3 above)
Comment (nirmal864) [reply Server/src/Models/Client.js]: removed ids

**Review by Pyxsys — APPROVED** [review]

**Comment by NathanZiri** [comment]
Looks good!

**Review by NathanZiri — APPROVED** [review]
Looks Good!

**Review by SekayZ — APPROVED** [review]
Looks good!

**Pull request merged**